### PR TITLE
add install animation

### PR DESCRIPTION
**Pull request by Pierregvx**
[Screencast from 28-06-2023 11:35:16.webm](https://github.com/massalabs/station/assets/90157528/9f0f33f4-e5bc-42f0-85df-8eb6dd24fd1f)

we can't see well, but the icon will pulse like this :  see [here](https://tailwindcss.com/docs/animation#pulse) 
author @pivilartisant

### CODE REVIEW

**Review by pivilartisant — COMMENTED** [review]
Can you make a little video to demonstrate in the pr description ?

**Comment by Pierregvx** [comment]
> Cna you make a little video to demonstrate in the pr description ?

image

**Comment by pivilartisant** [comment]
> > Cna you make a little video to demonstrate in the pr description ?
> 
> image

ok I think the bg needs to be the same as the plugin card @mazmassa do we need to change the ui-kit to do this ?

**Comment by mazmassa** [comment]
> ok I think the bg needs to be the same as the plugin card @mazmassa do we need to change the ui-kit to do this ?

I guess yes. BTW the horizontal position also needs a tiny refactor to be centered. Yes, is in the Ui-Kit.

**Comment by pivilartisant** [comment]
> > ok I think the bg needs to be the same as the plugin card @mazmassa do we need to change the ui-kit to do this ?
> 
> I guess yes. BTW the horizontal position also needs a tiny refactor to be centered. Yes, is in the Ui-Kit.

I think the best option would be a customBgClass ?

**Comment by Pierregvx** [comment]
[Screencast from 27-06-2023 14:58:26.webm](https://github.com/massalabs/station/assets/83134981/1422d4c7-7165-4eb9-bc96-ef88208ec358)

File: web/massastation/src/mirage/server.js [hunk 1]
```diff
@@ -159,7 +159,7 @@ function mockServer(environment = ENV.DEV) {
         return new Response(200, {});
       });
 
-      this.post('plugin-manager', (schema, request) => {
+      this.post('plugin-manager', { timing: 2000 }, (schema, request) => {
```
Comment (lilyanB) [inline web/massastation/src/mirage/server.js:162]: don't push this, I thinks

**Review by lilyanB — CHANGES_REQUESTED** [review]

File: web/massastation/src/mirage/server.js (hunk 1 above)
Comment (Pierregvx) [reply web/massastation/src/mirage/server.js:162]: why ?

**Comment by Pierregvx** [comment]
> > Screencast.from.27-06-2023.14.58.26.webm
> 
> uhm I dont think we should make the whole plugin card fade, but only the intsall icon... I'm sorry if that wasn't clear :/

Are you sure ? the user won't be able to access the other plugins then

File: web/massastation/src/mirage/server.js (hunk 1 above)
Comment (lilyanB) [reply web/massastation/src/mirage/server.js:162]: I thought we didn't change mirage. But okay

**Review by lilyanB — APPROVED** [review]

File: web/massastation/src/mirage/server.js (hunk 1 above)
Comment (Pierregvx) [reply web/massastation/src/mirage/server.js]: removed

File: web/massastation/src/pages/Store/StoreSection/StorePlugin.tsx [hunk 2]
```diff
@@ -30,7 +30,19 @@ function StorePlugin(props: StorePluginProps) {
...
+        customClass={isLoading ? 'animate-pulse blur-sm' : ''}
+        onClick={() => mutate({ params })}
+        disabled={isLoading}
+      >
+        <div
+          className={`rounded-full ${isLoading ? 'animate-pulse blur-sm' : ''}`}
+        >
+          <FiDownload />
+        </div>
+      </Button>
```
Comment (mazmassa) [inline web/massastation/src/pages/Store/StoreSection/StorePlugin.tsx:44]: I would prefer we extract to a file called `Loading` as we did for Search (and other) pages and here we just switch between them. isLoading ? <Loading/> : <...>

Comment (Pierregvx) [reply web/massastation/src/pages/Store/StoreSection/StorePlugin.tsx]: done

Comment (mazmassa) [reply web/massastation/src/pages/Store/StoreSection/StorePlugin.tsx]: Cool! Thank you! 🎉

File: web/massastation/src/pages/Store/StoreSection/StorePlugin.tsx [hunk 3]
```diff
@@ -9,6 +9,13 @@ interface StorePluginProps {
   plugin: IMassaStore;
   refetch: () => void;
 }
+
+const LoadingDownload = () => (
+  <div className={`rounded-full animate-pulse blur-sm`}>
+    <FiDownload />
+  </div>
+);
```
Comment (mazmassa) [inline web/massastation/src/pages/Store/StoreSection/StorePlugin.tsx:17]: I wont complain if you decide to keep as it is here but we, as a team, we said that we would use regular functions for that.

```suggestion
function LoadingDownload() {
  return ( 
    <div className={`rounded-full animate-pulse blur-sm`}>
      <FiDownload />
    </div>
  );
}
```

Comment (Pierregvx) [reply web/massastation/src/pages/Store/StoreSection/StorePlugin.tsx]: sorry about that, I just changed it

**Comment by pivilartisant** [comment]
@SlnPons can you Pre-QA this ? The spinner didn't seem to be the best option, so I suggested changing the icon style -> like making it blurry. If you don't like it we can discuss alternatives :)

File: web/massastation/src/mirage/server.js
Comment (mazmassa) [inline web/massastation/src/mirage/server.js:1]: I guess this is out of date since @maxime-carabina already changed. Check you rebase!

Comment (mazmassa) [reply web/massastation/src/mirage/server.js:1]: https://discord.com/channels/@me/1100786679486369844/1123540263126704129

**Pull request merged**